pkg/provisioningv2/rke2/planner/store.go

```diff
@@ -360,12 +360,12 @@ func assignAndCheckPlan(store *PlanStore, msg string, server *planEntry, newPlan
 		}
 		return ErrWaiting(fmt.Sprintf("starting %s", msg))
 	}
-	if !server.Plan.InSync {
-		return ErrWaiting(fmt.Sprintf("waiting for %s", msg))
-	}
 	if server.Plan.Failed {
 		return fmt.Errorf("operation %s failed", msg)
 	}
+	if !server.Plan.InSync {
+		return ErrWaiting(fmt.Sprintf("waiting for %s", msg))
```

Comment:
Out of curiosity why are you moving this?

Reply:
This was moved because I encountered situations where a plan could be not in sync, but also failed (IIRC when a plan's max-failures is changed from unlimited to a finite amount, but not the actual plan itself). This caused the plan to report out of sync, but it would never actually try again.